crates/ink/ir/src/ir/event/signature_topic.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use hex::decode_to_slice;
+use proc_macro2::{
+    Span,
+    TokenStream as TokenStream2,
+};
+use quote::ToTokens;
+use syn::spanned::Spanned;
+
+use crate::{
+    ast,
+    error::ExtError,
+    utils::extract_cfg_attributes,
+};
+
+/// The signature topic argument of an event variant.
+///
+/// Used as part of `ink::event` macro.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct SignatureTopicArg {
+    topic: [u8; 32],
+}
+
+impl SignatureTopicArg {
+    pub fn signature_topic(&self) -> [u8; 32] {
+        self.topic
+    }
+}
+
+impl From<&[u8; 32]> for SignatureTopicArg {
+    fn from(value: &[u8; 32]) -> Self {
+        Self { topic: *value }
+    }
+}
+
+impl TryFrom<&ast::MetaValue> for SignatureTopicArg {
+    type Error = syn::Error;
+
+    fn try_from(value: &ast::MetaValue) -> Result<Self, Self::Error> {
+        if let ast::MetaValue::Lit(lit) = value {
+            if let syn::Lit::Str(s) = lit {
```

Comment:
Since we are decoding the hex string here, we should rather use `Lit::BytStr` as we do for [selector](https://github.com/paritytech/ink/blob/bcbfa6697b5a6f59c4b27f3082dd6f6a20b0c236/crates/ink/ir/src/ir/selector.rs#L194).

Then can do `signature_topic = 0x1111111111111111111111111111111111111111111111111111111111111111`.

So we only support the byte literal and not strings.

Reply:
This wouldn't work since `0x` format is only for integers, and the max value is `0xffffffffffffffffffffffffffffffff`.
using `b"0x1111111111111111111111111111111111111111111111111111111111111111"` wouldn't also work because it parses characters as ASCI. Therefore, using raw strings allows to specify hashes directly, which can also be copy-pasted from the metadata of the the other contract.